### ISSUE TITLE
As an importer, I want to load a list of type material records from a file

### ISSUE THREAD

**Issue by LordFlashmeow** [SpeciesFileGroup/taxonworks#2363]
The DwC importer is designed to only create type material records for valid specimen names (when the scientific name matches the name parsed from the typeStatus field). This means that type specimen records for invalid names are not automatically generated, since we use the specimens' valid name during the import.

To work around this, there should be a batch import task for type materials, which can provide more thorough validation than is in the scope of the DwC importer.

It should match catalogNumbers to an existing `Identifier::Local::CatalogNumber`, and match the taxonomic name to an existing TW name. 

For example:
| catalog_number | taxon_name | type_type |
|---|---|---|
| CASENT0101169 | Crematogaster descarpentriesi | syntype |
| CASENT0101769 | Crematogaster adrepens | lectotype | 
| CASENT0101664 | Crematogaster emmae | holotype |

**Comment by mjy** [comment]
This is a new batch-loader.  If the behaviour is of only one type (see below) then we could add it using the old framework.  If we need more complicated options then perhaps we should wait and build it with the new batch load backend from the staged_imports branch?

This would be straightforward to build if we use the rules:

* There is one, and only one CollectionObject with a Identifier.cached = catalog_number (and only one identifier with this value, regardless of type)
* There is one and only one TaxonName with `cached` and/or `cached_original_combination` equal to `taxon_name`

@LordFlashmeow a) is the pattern above useful enough for you and b) how many records are you referencing for the ant data?

**Comment by LordFlashmeow** [comment]
Yes, that pattern matches our data. We have ~25k type records to import, and only 10k are for valid names (and can be parsed by the DwC importer). This leaves ~15k type records for invalid names, which this importer would have to work on. 

(We still need to import the 15k invalid names, see #2364)

**Comment by LordFlashmeow** [comment]
Maybe just flag rows that don't follow the rules (duplicate catalog number or multiple matching taxon names)?

Handling homonyms for this automatically is hard, but very unlikely/infrequent and can be resolved by hand.
